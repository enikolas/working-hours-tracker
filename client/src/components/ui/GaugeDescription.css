.GaugeDescription {
  font-size: 13px;
  color: #5a717f;
  display: block;
}

.GaugeDescription strong {
  color: #254356;
  margin-left: 4px;
}
