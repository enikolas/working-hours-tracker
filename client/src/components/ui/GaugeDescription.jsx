import React from 'react';
import PropTypes from 'prop-types';
import TimeDuration from 'time-duration';

import './GaugeDescription.css';

const formatValue = value => (new TimeDuration(value)).toString();

const GaugeDescription = ({ description, value }) => (
  <span className="GaugeDescription">
    {description} <strong>{formatValue(value)}</strong>
  </span>
);

export default GaugeDescription;

GaugeDescription.propTypes = {
  description: PropTypes.string.isRequired,
  value: PropTypes.number.isRequired
};
