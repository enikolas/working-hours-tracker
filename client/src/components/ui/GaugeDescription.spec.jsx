import React from 'react';
import { shallow } from 'enzyme';

import GaugeDescription from './GaugeDescription';

describe('GaugeDescription', () => {
	it('should render with a properly', () => {
		const wrapper = shallow(<GaugeDescription description="Some description" value={8 * 60} />);
		expect(wrapper).toMatchSnapshot();
	});
});
