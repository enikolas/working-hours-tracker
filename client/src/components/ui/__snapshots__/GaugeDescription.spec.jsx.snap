// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GaugeDescription should render with a properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GaugeDescription
    description="Some description"
    value={480}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Some description",
        " ",
        <strong>
          8:00
        </strong>,
      ],
      "className": "GaugeDescription",
    },
    "ref": null,
    "rendered": Array [
      "Some description",
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "8:00",
        },
        "ref": null,
        "rendered": "8:00",
        "type": "strong",
      },
    ],
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Some description",
          " ",
          <strong>
            8:00
          </strong>,
        ],
        "className": "GaugeDescription",
      },
      "ref": null,
      "rendered": Array [
        "Some description",
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "8:00",
          },
          "ref": null,
          "rendered": "8:00",
          "type": "strong",
        },
      ],
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
